src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+    { &gameplay_keep_Linkanim_00DA90, &gameplay_keep_Linkanim_00DA98 },
+    { &gameplay_keep_Linkanim_00D1D8, &gameplay_keep_Linkanim_00D1E0 },
+    { &gameplay_keep_Linkanim_00D1F0, &gameplay_keep_Linkanim_00D1E8 },
+};
+
+static EffectBlureInit2 sBlureInit = {
+    0, 8, 0, { 255, 255, 255, 255 }, { 255, 255, 255, 64 }, { 255, 255, 255, 0 }, { 255, 255, 255, 0 }, 4,
+    0, 2, 0, { 0, 0, 0, 0 },         { 0, 0, 0, 0 },
+};
+
+static EffectTireMarkInit sTireMarkInit = {
+    0,
+    63,
+    { 0, 0, 15, 100 },
+};
+
+static struct_80A40678 D_80A417E8[] = {
+    { func_80A3F080, func_80A3F09C }, { func_80A40098, func_80A40230 }, { func_80A3F62C, func_80A3F73C },
+    { func_80A3F8D4, func_80A3F9A4 }, { func_80A3F9E4, func_80A3FA58 }, { func_80A3FBCC, func_80A3FBE8 },
+    { func_80A3FDE4, func_80A3FE20 }, { func_80A3FF10, func_80A3FFD0 },
+};
+
+static struct_80A41828 D_80A41828[] = {
+    { 0, 0, 0 },  { 3, -2, 0 }, { 2, -1, 0 }, { 2, -1, 1 }, { 2, -1, 2 }, { 2, 1, 3 }, { 2, -2, 4 }, { 5, 1, 0 },
+    { 7, -1, 0 }, { 6, 2, 0 },  { 4, 4, 0 },  { 1, 1, 0 },  { 1, 1, 0 },  { 1, 2, 0 }, { 1, 1, 0 },  { 1, 1, 0 },
+    { 1, 2, 0 },  { 1, 3, 0 },  { 1, 1, 0 },  { 1, 2, 0 },  { 1, 1, 0 },  { 0, 0, 0 },
+};
+
+static struct_D78 D_80A41854[] = {
+    { 4, 0, 0x2B25 }, { 1, 0, 0x2969 }, { 3, 1, 0x296A },  { 1, 0, 0x296B }, { 5, 1, 0x0000 }, { 8, 0, 0x0000 },
+    { 4, 0, 0x2976 }, { 6, 0, 0x2977 }, { 7, 10, 0x2978 }, { 4, 1, 0x0000 }, { 4, 0, 0x2968 }, { 4, 0, 0x297A },
+    { 1, 0, 0x145D }, { 1, 0, 0x145E }, { 5, 1, 0x145F },  { 1, 0, 0x145F }, { 5, 0, 0x0000 }, { 4, 0, 0x1460 },
+    { 4, 0, 0x145C }, { 4, 0, 0x2913 }, { 4, 0, 0x1465 },
+};
+
+s32 D_80A41D20;
+s32 D_80A41D24;
+static Input sInput;
+f32 D_80A41D40;
+s16 D_80A41D44;
+s32 D_80A41D48;
+Vec3f D_80A41D50;
+s32 D_80A41D5C;
+s32 D_80A41D60;
+s32 D_80A41D64;
+s32 D_80A41D68;
+Vec3f* D_80A41D6C;
+
+s32 func_80A3E7E0(EnTest3* this, EnTest3ActionFunc actionFunc) {
+    if (actionFunc == this->unk_D94) {
+        return false;
+    } else {
+        this->unk_D94 = actionFunc;
+        this->unk_D8A = 0;
+        this->unk_D88 = 0;
+        return true;
+    }
+}
+
+s32 func_80A3E80C(EnTest3* this, PlayState* play, s32 arg2) {
+    s32 pad;
+
+    D_80A4168C[arg2].unk_0(this, play);
+    if (D_80A4168C[arg2].actionFunc == NULL) {
+        // D_80A4168C[arg2].actionFunc is always NULL
+        return false;
+    } else {
+        func_80A3E7E0(this, D_80A4168C[arg2].actionFunc);
+        return true;
+    }
+}
+
+s32 func_80A3E870(EnTest3* this, PlayState* play) {
+    return true;
+}
+
+s32 func_80A3E884(EnTest3* this, PlayState* play) {
+    return false;
+}
+
+s32 func_80A3E898(EnTest3* this, PlayState* play) {
+    u16 textId = this->unk_D78->textId;
+
+    if ((this->unk_D78->unk_0 == 4) && (gSaveContext.save.weekEventReg[51] & 8)) {
+        func_80151BB4(play, 2);
+    }
+    if (textId == 0xFFFF) {
+        func_801477B4(play);
+    } else if (textId) { // != 0
```

Comment:
It is, though the purpose of the comment was a reminder that it's checking against 0, since every other textId check in the function checks against a literal.